include/libdnf/advisory/advisory.hpp

```diff
@@ -0,0 +1,135 @@
+#ifndef LIBDNF_ADVISORY_ADVISORY_HPP
+#define LIBDNF_ADVISORY_ADVISORY_HPP
+
+#include <libdnf/rpm/solv_sack.hpp>
+
+#include <vector>
+
+
+namespace libdnf {
+class Base;
+}
+
+namespace libdnf::advisory {
+
+class AdvisoryCollection;
+class AdvisoryReference;
+
+typedef libdnf::rpm::PackageId AdvisoryId;
+
+/// Type of advisory
+enum class AdvisoryType : uint32_t {
+    UNKNOWN = (1 << 0),
+    SECURITY = (1 << 1),
+    BUGFIX = (1 << 2),
+    ENHANCEMENT = (1 << 3),
+    NEWPACKAGE = (1 << 4)
+};
+
+/// Type of advisory reference
+enum class AdvisoryReferenceType : uint32_t {
+    UNKNOWN = (1 << 0),
+    BUGZILLA = (1 << 1),
+    CVE = (1 << 2),
+    VENDOR = (1 << 3),
+};
+
+inline AdvisoryReferenceType operator|(AdvisoryReferenceType lhs, AdvisoryReferenceType rhs) {
+    return static_cast<AdvisoryReferenceType>(
+        static_cast<std::underlying_type<AdvisoryReferenceType>::type>(lhs) |
+        static_cast<std::underlying_type<AdvisoryReferenceType>::type>(rhs));
+}
+
+inline AdvisoryReferenceType operator&(AdvisoryReferenceType lhs, AdvisoryReferenceType rhs) {
+    return static_cast<AdvisoryReferenceType>(
+        static_cast<std::underlying_type<AdvisoryReferenceType>::type>(lhs) &
+        static_cast<std::underlying_type<AdvisoryReferenceType>::type>(rhs));
+}
+
+/// An advisory, represents advisory used to track security updates
+class Advisory {
+public:
+    using Type = AdvisoryType;
+
+    /// Get name of this advisory.
+    ///
+    /// @return Name of this advisory as std::string.
+    std::string get_name() const;
+
+    /// Get severity of this advisory.
+    ///
+    /// @return Severity of this advisory as std::string.
+    std::string get_severity() const;
+
+    /// Get type of this advisory.
+    ///
+    /// @return Type of this advisory.
+    Type get_type() const;
+
+    /// Get AdvisoryId.
+    ///
+    /// @return AdvisoryId of this advisory.
+    AdvisoryId get_id() const;
+
+    /// Get all references of specified type from this advisory.
+    ///
+    /// @param ref_type     What type of references to get. If not specified gets all types.
+    /// @return Vector of AdvisoryReference objects.
+    std::vector<AdvisoryReference> get_references(
+        AdvisoryReferenceType ref_type = AdvisoryReferenceType::BUGZILLA | AdvisoryReferenceType::CVE |
+                                         AdvisoryReferenceType::VENDOR) const;
+
+    /// Get all collections from this advisory.
+    ///
+    /// @return Vector of AdvisoryCollection objects.
+    std::vector<AdvisoryCollection> get_collections() const;
+
+    /// Check whether at least one collection from this advisory is applicable.
+    ///
+    /// @return True if applicable, False otherwise.
+    bool is_applicable() const;
+
+    /// Construct the Advisory object
+    ///
+    /// @param base   Reference to Base instance.
+    /// @param id     AdvisoryId into libsolv pool.
+    /// @return New Advisory instance.
+    Advisory(Base & base, AdvisoryId id);
```

Comment:
I suggest to move the constructor and destructor upward as we have it in other classes.